includes/admin/meta-boxes/class-wc-meta-box-product-data.php

```diff
@@ -220,6 +218,54 @@ public static function prepare_attributes( $data = false ) {
 	}
 
 	/**
+	 * Prepare default attributes.
+	 * @param  array $all_attributes
+	 * @return array
+	 */
+	public static function prepare_default_attributes( $all_attributes ) {
+		$attributes = array();
+
+		if ( $all_attributes ) {
+			foreach ( $all_attributes as $attribute ) {
+				if ( $attribute->get_variation() ) {
+					$attribute_key                = sanitize_title( $attribute->get_name() );
+					$post_key                     = 'default_attribute_' . $attribute_key;
+					$value                        = isset( $_POST[ $post_key ] ) ? stripslashes( $_POST[ $post_key ] ) : '';
+					$value                        = $attribute->is_taxonomy() ? sanitize_title( $value ) : wc_clean( $value ); // Don't use wc_clean as it destroys sanitized characters in terms.
+					$attributes[ $attribute_key ] = $value;
+				}
+			}
+		}
+
+		return $attributes;
+	}
+
+	/**
+	 * Prepare attributes for a specific variation.
+	 * @param  array $all_attributes
+	 * @param  int $variation_id
+	 * @param  int $i
+	 * @return array
+	 */
+	private static function prepare_variation_attributes( $all_attributes, $variation_id, $i ) {
```

Comment:
Since this function is pretty similar to the one above it, could we combine them? I/e, set the the `post_key` differently if `$variation_id` is set, using `$i`  if it's present, etc. (Which could be renamed since we usually see $i in for loops, and it's not super descriptive here.)